### PR TITLE
Override BigQueryIO.Write with native sink for Dataflow runner

### PR DESCRIPTION
**Pull request by peihe**

### CODE REVIEW

**Comment by peihe** [comment]
R: @dhalperi

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java [hunk 1]
```diff
@@ -2059,6 +2068,104 @@ public PDone apply(PCollection<T> input) {
...
+            .apply(new BatchBigQueryIONativeWrite(transform));
+      }
+    }
+  }
+
+  /**
+   * This {@link PTransform} is used by the {@link DataflowPipelineTranslator} as a way
+   * to provide the native definition of the BigQuery sink.
+   */
+  private static class BatchBigQueryIONativeWrite extends PTransform<PCollection<TableRow>, PDone> {
```
Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java]: I think this replaces BigQueryIOTranslator, right? Remove those?

Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java]: (For write only)

Comment (peihe) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java]: done

**Comment by dhalperi** [comment]
LGTM. Can you:

- [ ] confirm that when you build from github, we get native io by default
- [ ] confirm that when you build from github, the experiment successfully switches back to BigQueryIO.Write
- [ ] squash, so we can merge via the PR?

**Comment by peihe** [comment]
Done

Native sink:
https://pantheon.corp.google.com/dataflow/job/2016-04-15_14_42_38-12279249624446294392?project=google.com:clouddfe

Sink with experiments flag:
http://shortn/_zFDnqUQeSghttps://pantheon.corp.google.com/dataflow/job/2016-04-15_15_22_55-610896668165684866?project=google.com:clouddfe

**Comment by peihe** [comment]
There is a bug in BigQuery Custom Sink:
the pipeline will fail if there is double '/' in the concatenated tempPath.

I think we want to merge this PR, and I will send out a fix for the tempPath.

**Comment by dhalperi** [comment]
Awesome!

**Pull request merged**